src/libraries/System.Text.RegularExpressions/tests/RegexSRMTests.cs

```diff
@@ -37,6 +37,43 @@ public void TestBoundary()
             Assert.True(Regex.IsMatch(" AB\u200dCD ", @"\b\w+\b"));
         }
 
+        public static IEnumerable<object[]> TestLazyLoops_Data()
+        {
+            foreach (var options in new RegexOptions[] { RegexOptions.Singleline, RegexOptions.Compiled | RegexOptions.Singleline, DFA | RegexOptions.Singleline })
+            {
+                yield return new object[] { options, "seq 012 of 3 digits", @"\W.*?\D", true, 3, 5 };
+                yield return new object[] { options, "seq 012 of 3 digits", @"\W.+?\D", true, 3, 5 };
+                yield return new object[] { options, "seq 012 of 3 digits", @"\W.{1,7}?\D", true, 3, 5 };
+                yield return new object[] { options, "seq 012 of 3 digits", @"\W.{1,2}?\D", true, 7, 3 };
+                yield return new object[] { options, "digits:0123456789", @"\W.*?\b", true, 6, 1 };
+                yield return new object[] { options, "e.g:abc", @"\B.*?\B", true, 5, 0 };
+                yield return new object[] { options, "e.g:abc", @"\B\W+?", false, 0, 0 };
+            }
+            yield return new object[] { RegexOptions.Singleline, "e.g:abc", @"\B\W*?", true, 5, 0 };
+            yield return new object[] { RegexOptions.Compiled | RegexOptions.Singleline, "e.g:abc", @"\B\W*?", true, 5, 0 };
```

Comment:
when we fix the DFA bug we just adjust these two lines so they're inside the loop above, right? we would not have the TestLazyLoops_ActiveIssue test.

if so maybe that test is not worth having, and instead you just want to make sure there's an issue open to fix this. you could put the link to the issue here if you like.

Reply:
in fact, should these just go into Match_Basic_TestData? no need for a new test. all variations can get merged in there.

Reply:
I can also just leave the test where it was and let it fail until fixed. I was planning to do it next actually. Most probably part of this pr. Then I'll just put it back where it was -- change the TestLazyLoops_Data back to what it was initially when I added the test.

Reply:
I'll move the test to Match_Basic_TestData.